### DISCUSSION TITLE
Randomized grain selection

### DISCUSSION THREAD

**Discussion by GeoGingerBeard** [mtex-toolbox/mtex#1940] [Ask Anything] [answered]
Hello Mtex community,

I am trying to compare textural indices of several samples with a lot of variation in the total number of grains. To do this, I would like to randomly sample ~300 grains from a larger group of ~2000 grains for the purpose of recalculating  the J index and M index. I know I can select grains using the grains.selected function or through the list by something like "grains(1:300)" but is there a way to create a smaller list of random grains from a larger list. Your help is very much appreciated.

-Ben

**Comment by zmichels** [comment]
Hi,

I think if I were doing this, I would use the randi() function of Matlab on the grainIds. This allows you to specify the maximum possible integer and the number of random integers to be output.

Example from Mathworks online documentation:
X = randi(imax,n) 
returns an n-by-n matrix of pseudorandom integers drawn from the discrete uniform distribution on the interval [1,imax].

Then you'd follow that with something like:
randGrains = grains(X)

Link for more details:
https://www.mathworks.com/help/matlab/ref/randi.html

**Comment by ralfHielscher** [comment, marked as answer]
You can do
```matlab
disreteSample(grains,300)
```
to randomly select 300 out of your grains.

Ralf.
